### PR TITLE
Issue #SB-2157 feat: collection editor creation"

### PR DESCRIPTION
**Pull request by revathitarento**
Issue #SB-2157 feat: collection editor creation"

### CODE REVIEW

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateCollectionComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+
+  it('should call savemeta method', inject([EditorService, UserService], (editorService, userService) => {
+
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.spec.ts]: add assertion

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { ISubscription } from 'rxjs/Subscription';
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.spec.ts]: remove

Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.spec.ts]: Done

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import { async, ComponentFixture, TestBed, inject, fakeAsync, tick } from '@angular/core/testing';
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { CreateCollectionComponent } from './create-collection.component';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { ReactiveFormsModule } from '@angular/forms';
+import { SuiModule } from 'ng2-semantic-ui';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { Ng2IziToastModule } from 'ng2-izitoast';
+import { SuiModalService, TemplateModalConfig, ModalTemplate } from 'ng2-semantic-ui';
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.spec.ts]: remove

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
+import { UserService } from '@sunbird/core';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { SuiModule } from 'ng2-semantic-ui/dist';
+import { SuiModalService, TemplateModalConfig, ModalTemplate } from 'ng2-semantic-ui';
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: remove

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  public resourceService: ResourceService;
+  /**
+   * To make inbox API calls
+   */
+  private editorService: EditorService;
+
+  /**
+   * userProfile is of type userprofile interface
+   */
+  public userProfile: any;
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: it should not be any

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
+import { UserService } from '@sunbird/core';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: remove

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+import { SuiModule } from 'ng2-semantic-ui/dist';
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: remove

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+  /**
+   * userProfile is of type userprofile interface
+   */
+  public userProfile: any;
+
+  public userForm: FormGroup;
+  public formBuilder: FormBuilder;
+
+  constructor(public modalService: SuiModalService,
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: remove modalService: SuiModalService,

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+  private editorService: EditorService;
+
+  /**
+   * userProfile is of type userprofile interface
+   */
+  public userProfile: any;
+
+  public userForm: FormGroup;
+  public formBuilder: FormBuilder;
+
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: documentation

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+    this.showLoader = true;
+      const requestBody = {
+      name: this.userForm.value.name ? this.userForm.value.name : 'Untitled Collection',
+      description: this.userForm.value.description,
+      creator: this.userProfile.firstName + ' ' + this.userProfile.lastName,
+      createdBy: this.userProfile.id,
+      organisationIds: this.userProfile.organisationIds,
+      createdFor: this.userProfile.organisationIds,
+      userRoles: this.userProfile.userRoles,
+      mimeType: 'application/vnd.ekstep.content-collection',
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: move this to config

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+      contentType: 'Collection'
+    };
+    return requestBody;
+  }
+
+// Create colletion crates the content Id
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: follow doc syntax

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+      const framework = 'framework';
+      this.router.navigate(['/workspace/content/edit/collection', res.result.content_id, type, state, framework]);
+    }, err => {
+      this.toasterService.error(this.resourceService.messages.emsg.m0005);
+    });
+
+  }
+
+
+  goToCreate() {
```
Comment (anoophm) [inline src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: doc

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.spec.ts (hunk 3 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.spec.ts]: done

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts (hunk 4 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: done

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts (hunk 5 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: done

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts (hunk 6 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: done

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts (hunk 8 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: done

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts (hunk 9 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: done

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts (hunk 11 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: done

File: src/app/migration/src/src/app/modules/workspace/services/editors/editor.service.spec.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
+
+import { async, ComponentFixture, TestBed, inject, fakeAsync, tick } from '@angular/core/testing';
+// Import NG testing module(s)
```
[the remark this answers is not in the document]
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/services/editors/editor.service.spec.ts]: Done

File: src/app/migration/src/src/app/modules/workspace/services/editors/editor.service.spec.ts [hunk 14]
```diff
@@ -0,0 +1,41 @@
+
+import { async, ComponentFixture, TestBed, inject, fakeAsync, tick } from '@angular/core/testing';
```
[the remark this answers is not in the document]
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/services/editors/editor.service.spec.ts]: done

File: src/app/migration/src/src/app/modules/workspace/services/editors/editor.service.spec.ts [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+// Import NG testing module(s)
+import { HttpClientModule } from '@angular/common/http';
+// Import services
+// Rxjs
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
```
[the remark this answers is not in the document]
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/services/editors/editor.service.spec.ts]: done

File: src/app/migration/src/src/app/modules/workspace/services/editors/editor.service.ts [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+        this.contentService = contentService;
+        this.baseUrl = this.config.urlConFig.URLS.CONTENT_PREFIX;
+    }
+
+/**
+ * Create content Id for the editor
+ * @param req
+ */
+    create(req) {
+        console.log('req', req);
```
[the remark this answers is not in the document]
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/services/editors/editor.service.ts]: done

File: src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts (hunk 12 above)
Comment (revathitarento) [reply src/app/migration/src/src/app/modules/workspace/components/content-creation-forms/create-collection/create-collection.component.ts]: Done

**Pull request merged**